functions/src/plugins/rerun-circleci.ts

```diff
@@ -98,4 +97,73 @@ ${error.message}
     const repositoryConfig = await this.getAppConfig(context);
     return repositoryConfig.rerunCircleCI;
   }
+
+  /**
+   * Get the CircleCI workflow id of the first discovered CircleCI status in the statuses.  Since
+   * only one workflow is run on each PR, all CircleCI statuses will track back to the same
+   * workflow id.
+   */
+  private async getCircleCiWorkflowIdForPullRequest(context: Context) {
+    /** The target url of the discovered CircleCI status. */
+    let targetUrl: string;
+    /** The pull request which triggered the bot action. */
+    const pullRequest: Github.PullRequestsGetResponse = context.payload.pull_request;
+    /** The owner and repository name. */
+    const {owner, repo} = context.repo();
+    /** The list of statuses for the latest ref of the PR. */
+    const {statuses} = (await context.github.repos.getCombinedStatusForRef({
+      owner, repo, ref: pullRequest.head.ref
+    })).data;
+
+    for (const status of statuses) {
+      if (status.context.startsWith('ci/circleci:')) {
+        targetUrl = status.target_url;
+        break;
+      }
+    }
+
+    if (targetUrl === undefined) {
+      throw Error('No status for a CircleCI workflow was found on the pull request to be rerun.');
+    }
+
+    /**
+     * The matcher results of the regex to select the job ID of the job which the status represents.
+     */
+    const jobIdMatcher = targetUrl.match(`https://circleci.com/gh/${owner}/${repo}/(\d+)\?`);
```

Comment:
I don't think this string is converted to the RegExp you intend it to be converted :grin:
What you have now will basically match `https://circleci.com/gh/${owner}/${repo}/` followed by zero or more `d` letters :grin:

You were probably going for something like:

```suggestion
    const jobIdMatcher = targetUrl.match(`https://circleci\\.com/gh/${owner}/${repo}/(\\d+)\\?`);
```